datafusion/functions/src/string/common.rs

```diff
@@ -418,6 +448,148 @@ impl StringArrayBuilder {
     }
 }
 
+pub(crate) struct StringViewArrayBuilder {
+    builder: StringViewBuilder,
+    block: String
+}
+
+impl StringViewArrayBuilder {
```

Comment:
`StringViewArrayBuilder` works but it almost feels "hacky"

Reply:
I filed https://github.com/apache/arrow-rs/issues/6347 to discuss adding an API like this in arrow as well